### PR TITLE
MTP-2801 ChapterMenuSubChapter: Remove xstate and add tests

### PR DESCRIPTION
**Pull request by kajsaeggum**
- Removed use of xstate from component
- Changed default loadJs prop to false
- Added component tests
- Also fixed components tests for unrelated component CheckboxWithSubSets since they had commented out checks for the same reason relating to use of svelte transition.

### CODE REVIEW

File: mt-kit/core/svelte/src/lib/svelte/components/ChapterMenuSubChapter.test.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    subChapters: subChapters,
+    ariaLabel: 'view sub chapters',
+    loadJs: true
+  }
+
+  /**
+   * Fix so svelte-transitions don´t get stuck in transition,
+   * i.e. never completely hide element, when running tests
+   * Ref: https://github.com/testing-library/svelte-testing-library/issues/206#issuecomment-1470158576
+   */
```
Comment (lha047) [inline mt-kit/core/svelte/src/lib/svelte/components/ChapterMenuSubChapter.test.ts:23]: Nice :D Like the the comment explaining why this is needed

File: mt-kit/core/svelte/src/lib/svelte/components/ChapterMenuSubChapter.svelte [hunk 2]
```diff
@@ -1,27 +1,30 @@
...
 
   if (loadJs) {
-    onMount(() => send('MOUNTED'))
+    onMount(() => {
+      onServer = false
+      isOpen = false
+    })
+  }
+
+  const handleButtonClick = (): void => {
```
Comment (lha047) [inline mt-kit/core/svelte/src/lib/svelte/components/ChapterMenuSubChapter.svelte:26]: Would prefere to use function expressions as the rest of the code has this style. But we can discuss :)

File: mt-kit/core/svelte/src/lib/svelte/components/ChapterMenuSubChapter.test.ts (hunk 1 above)
Comment (kajsaeggum) [reply mt-kit/core/svelte/src/lib/svelte/components/ChapterMenuSubChapter.test.ts:23]: Maybe I should move it to setuptest.js so it's "fixed" globally 🤔

File: mt-kit/core/svelte/src/lib/svelte/components/ChapterMenuSubChapter.svelte (hunk 2 above)
Comment (kajsaeggum) [reply mt-kit/core/svelte/src/lib/svelte/components/ChapterMenuSubChapter.svelte:26]: I fix :)

File: mt-kit/core/svelte/setuptest.js [hunk 3]
```diff
@@ -19,3 +19,12 @@ const ResizeObserverMock = vi.fn(() => ({
 }))
 
 vi.stubGlobal('ResizeObserver', ResizeObserverMock)
+
+/**
+ * Fix so svelte-transitions don´t get stuck in transition,
+ * i.e. never completely hide element, when running tests
+ * Ref: https://github.com/testing-library/svelte-testing-library/issues/206#issuecomment-1470158576
+ */
+vi.stubGlobal('requestAnimationFrame', fn => {
```
Comment (lha047) [inline mt-kit/core/svelte/setuptest.js:28]: 👍

**Review by lha047 — APPROVED** [review]
Good job 👍 
You can add this version to the web project

**Pull request merged**